### ISSUE TITLE
T1 poles with optional fuse

### ISSUE THREAD

**Issue by genestrashcan** [age-series/ElectricalAge#170]
New Item: add a t1 pole with transformer with a built in 13kv fuse

this would be useful to prevent a short circuit on grid lines from blowing up your whole grid.

it should coded so if a short circuit happens that you will hear the fuse instantly pop from far away because in real life when a fuse on a power pole blows it's louder than a shotgun.

**Comment by GrimmDemon** [comment]
Wouldn't it make sense to use a resetable fuse switch instead, we have one on the transformer pole near our house which steps the voltage down to usable 240V at 50hz. I coincidentally live close to an electrical substation which transforms electrical grid to lower voltage to be transformed by a further two tramsformer boxes, including the one mounted on the street lamp pole.
The noise generated by a cable explosion is roughly similar to TNT exploding, in the game of course, the sound of that can be heard from a small distance, but quite softly. Minecraft works by loading chunks, so it won't really be possible if you want the sound of the fuse popping from, say, 3 or 4 chunks away. The average explosion can be heard from 64 blocks away, that's about a chunk. People can try and make minecraft as realistic as possible, but it really makes it harder for a computer to render it, especially if you don't have many CPU cores. Even the sound of a nuke exploding in game using HBM's nuclear tech mod doesn't go further than two chunks, albeit excessively loud when you are even at the edge of the sound cutting out. I stopped using that mod although because it was extremely unstable, plus my base kept on being struck by falling meteorites, not a fan of my hard work being randomly destroyed.

**Comment by jrddunbr** [comment]
From personal experience, it's about as loud as a gunshot.

The fuses on poles are just fuses, not circuit breakers. That said, they are installed by clipping in the bottom hinge and then pushing up to complete the connection. They are going to be consumables.

The Minecraft sound engine makes it a pain for things to be heard far away. It requires the chunk to be loaded firstly, so over 4 chunks away is infeasible.

**Comment by GrimmDemon** [comment]
I didn't really think of it like that, I can imagine the arc caused by an electrical breaker running 13.2kV through it, it would almost immediately destroy the circuit breaker. The sound caused by high voltage wires colliding and shorting are loud, but personally, a gun shot is reasonably louder, unless it's a pistol. The noise will most probably be caused by the arc and the colliding electrons, and the flash is rather blinding, think of an arc welder, but several times brighter.

**Comment by jrddunbr** [comment]
Just keep in mind that 13kV is not the only voltage people are going to be running on poles.

There are circuit breakers designed for voltages like that, but they are prohibitively expensive to place on every pole ever. Fuses are far easier to make, and honestly replacing them is not hard.

**Comment by ellingtonvengaboysmuseum** [comment]
make the sound only heard within 1 chunk...

This video shows how loud one of these things is when it blows

https://www.youtube.com/watch?v=dXaoCfGPHGo

**Comment by jrddunbr** [comment]
stop changing the title.

**Comment by ellingtonvengaboysmuseum** [comment]
@jrddunbr - i may have come up with a idea on how to make the fuse exploding sound work from far away...

you will need several shot gun sounds for this

one will be the one for being very close to the fuse (within a chunk radius)
the other two would be sounds of a distant gun shot, one louder than the other

then you would then figure out how to write the code so that the fuse would check if the player was within a 30 block radius of it for the first sound of the closer distant gun shot and play the sound at the current players position if the player was above ground level and not down in a cave etc and have the area the sound is played be large enough so it's heard if the player is moving around

you would do the same thing for the second sound and have it check if the player was between 30 - 100 blocks away from the fuse popping and play the second distant gun shot sound at the players position above ground.

**Comment by jrddunbr** [comment]
The sound engine is a bit complex. I'm not going to explain here, but I don't think we need more than one sound

**Comment by GrimmDemon** [comment]
I agree. What you're proposing here is extremely complex to code. Honestly, minecraft is a simple game, I would probably just leave it like that instead of it requiring more cpu power for chunks that aren't loaded at that moment, I struggle enough to have more than 8 chunks at a time loaded in a single server.

**Comment by jrddunbr** [comment]
This issue is being closed now since there will be no more feature updates in Electrical Age for Minecraft 1.7.10.
If the issue is relevant to 1.12, it will be re-opened there at a later date.
